### PR TITLE
feat: add manifest v3 to the build flags

### PR DESCRIPTION
**Pull request by Jack-Works**
I'm going to start my very first exploration of the Web Extension Manifest V3. The first step is to add a new build target for it.

This PR introduces a new package `@dimensiondev/maskbook-cli-dev-or-build` (private) in our monorepo to simplify the over-heated `start:*` and `build:*` commands.

See: #365

### CODE REVIEW

File: scripts/ci-build.ts [hunk 1]
```diff
@@ -3,25 +3,25 @@ import { BUILD_PATH, run } from './utils'
...
     const target = types[filename]
-    console.log('#', 'Building for target:', target)
+    console.log('#', 'Building for target:', ...target)
     if (filename === 'chromium') {
         // chromium doesn't have it's own changes yet.
         // just copying base version is acceptable
         run(undefined, 'cp', '-v', 'Maskbook.base.zip', 'Maskbook.chromium.zip')
     } else {
-        run(undefined, 'npx', 'build', target)
+        run(undefined, 'npx', 'build', ...target)
```
Comment (septs) [inline scripts/ci-build.ts:24]: `npx build ?`

Comment (Jack-Works) [reply scripts/ci-build.ts:24]: Yes, the new package provides 2 new commands in the package level. `dev` and `build`

File: .github/workflows/e2e.yml [hunk 2]
```diff
@@ -37,7 +37,7 @@ jobs:
         run: npm ci
 
       - name: Build
-        run: npm run build:e2e
+        run: npm run build E2E
```
Comment (septs) [inline .github/workflows/e2e.yml:40]:
```suggestion
        run: npx build E2E
```

File: packages/cli-dev-build/main.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        args = [...flags, preset]
+        const { confirm } = await inquirer.prompt({
+            type: 'confirm',
+            name: 'confirm',
+            message: `Command is: "npx ${mode === 'dev' ? 'dev' : 'build'} ${args.join(' ')}". Is that OK?`,
+        })
+        if (!confirm) return
+    }
+
+    let command = mode === 'dev' ? 'serve --mode development' : '--mode production'
```
Comment (septs) [inline packages/cli-dev-build/main.js:32]: why not is array?

File: packages/cli-dev-build/main.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    args.forEach((flag) => {
+        command += ' --env ' + flag
+        if (!knownFlags.includes(flag)) throw new TypeError('Unknown flag ' + flag)
+    })
+    spawn('webpack', command.split(' '), { stdio: 'inherit', shell: true })
```
Comment (septs) [inline packages/cli-dev-build/main.js:37]:
```suggestion
    spawn('webpack', command, { stdio: 'inherit', shell: true })
```

File: packages/cli-dev-build/main.js (hunk 3 above)
Comment (septs) [reply packages/cli-dev-build/main.js:32]:
```suggestion
    let command = ['serve', '--mode', mode === 'dev' ? 'development' : 'production']
```

File: packages/cli-dev-build/main.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+            type: 'confirm',
+            name: 'confirm',
+            message: `Command is: "npx ${mode === 'dev' ? 'dev' : 'build'} ${args.join(' ')}". Is that OK?`,
+        })
+        if (!confirm) return
+    }
+
+    let command = mode === 'dev' ? 'serve --mode development' : '--mode production'
+    args.forEach((flag) => {
+        command += ' --env ' + flag
```
Comment (septs) [inline packages/cli-dev-build/main.js:34]:
```suggestion
        command.push('--env', flag)
```

File: packages/cli-dev-build/package.json [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+  "name": "@dimensiondev/maskbook-cli-dev-or-build",
+  "private": true,
+  "bin": {
+    "build": "./build.js",
+    "dev": "./dev.js"
+  },
+  "dependencies": {
+    "inquirer": "^7.3.3"
+  },
+  "devDependencies": {}
```
Comment (septs) [inline packages/cli-dev-build/package.json:11]:
```suggestion
```

**Review by septs — CHANGES_REQUESTED** [review]

File: packages/cli-dev-build/package.json [hunk 7]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "@dimensiondev/maskbook-cli-dev-or-build",
+  "private": true,
+  "bin": {
+    "build": "./build.js",
+    "dev": "./dev.js"
+  },
+  "dependencies": {
+    "inquirer": "^7.3.3"
+  },
```
Comment (Jack-Works) [inline packages/cli-dev-build/package.json:10]:
```suggestion
  }
```

**Comment by Jack-Works** [comment]
> Too long with no output (exceeded 10m0s): context deadline exceeded

Strange

**Comment by Jack-Works** [comment]
oh wait it becomes dev mode instead of prod mode

**Pull request merged**